### PR TITLE
Added rules tests, reworked  Iroha integration test

### PR DESCRIPTION
**Pull request by Mingela**
Any tests concerns and suggestions are appreciated

### CODE REVIEW

**Comment by Mingela** [comment]
Let it be the first iteration of testing

File: src/test/java/iroha/validation/rules/RulesTest.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
+package iroha.validation.rules;
```
Comment (dolgopolovwork) [inline src/test/java/iroha/validation/rules/RulesTest.java]: How about `given-when-then` comments structure?

File: src/test/java/iroha/validation/rules/RulesTest.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+  private void initTransferTxVolumeTest() {
+    asset = "asset";
+    rule = new TransferTxVolumeRule(asset, BigDecimal.TEN);
+    // transfer mock
+    transaction = mock(Transaction.class, RETURNS_DEEP_STUBS);
+    Command command = mock(Command.class);
+    transferAsset = mock(TransferAsset.class);
+
+    when(transaction
```
Comment (dolgopolovwork) [inline src/test/java/iroha/validation/rules/RulesTest.java]: Very strange formatting

File: src/test/java/iroha/validation/rules/RulesTest.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+import org.junit.jupiter.api.Test;
+
+class RulesTest {
+
+  private String asset;
+  private Transaction transaction;
+  private TransferAsset transferAsset;
+  private Rule rule;
+
+  private void initTransferTxVolumeTest() {
```
Comment (dolgopolovwork) [inline src/test/java/iroha/validation/rules/RulesTest.java]: You can mark this method with `@Before` annotation if you want to run this method before every test.

**Pull request merged**